lua/nvim-treesitter/query.lua

```diff
@@ -61,9 +61,16 @@ local function runtime_queries(lang, query_name)
   return api.nvim_get_runtime_file(string.format('queries/%s/%s.scm', lang, query_name), true) or {}
 end
 
+local query_files_cache = {}
 function M.has_query_files(lang, query_name)
-  local files = runtime_queries(lang, query_name)
-  return files and #files > 0
+  if not query_files_cache[lang] then
+    query_files_cache[lang] = {}
+  end
+	if query_files_cache[lang][query_name] == nil then
```

Comment:
```suggestion
	if not query_files_cache[lang][query_name] then
```

Reply:
It should be a check for `nil`, since if it is `false` we also don't need to check

Reply:
either we already checked, or the file esxists, or it doesn;t exist

Reply:
Oh yeah I see